core/src/main/java/org/geysermc/geyser/GeyserImpl.java

```diff
@@ -385,34 +381,36 @@ private void startInstance() {
                 if (parsedPort < 1 || parsedPort > 65535) {
                     throw new NumberFormatException("The broadcast port must be between 1 and 65535 inclusive!");
                 }
-                config.getBedrock().setBroadcastPort(parsedPort);
+                config.bedrock().broadcastPort(parsedPort);
                 logger.info("Broadcast port set from system property: " + parsedPort);
             } catch (NumberFormatException e) {
                 logger.error(String.format("Invalid broadcast port from system property: %s! Defaulting to configured port.", broadcastPort + " (" + e.getMessage() + ")"));
             }
         }
 
         // It's set to 0 only if no system property or manual config value was set
-        if (config.getBedrock().broadcastPort() == 0) {
-            config.getBedrock().setBroadcastPort(config.getBedrock().port());
+        if (config.bedrock().broadcastPort() == 0) {
+            config.bedrock().broadcastPort(config.bedrock().port());
         }
 
-        String remoteAddress = config.getRemote().address();
-        // Filters whether it is not an IP address or localhost, because otherwise it is not possible to find out an SRV entry.
-        if (!remoteAddress.matches(IP_REGEX) && !remoteAddress.equalsIgnoreCase("localhost")) {
-            String[] record = WebUtils.findSrvRecord(this, remoteAddress);
-            if (record != null) {
-                int remotePort = Integer.parseInt(record[2]);
-                config.getRemote().setAddress(remoteAddress = record[3]);
-                config.getRemote().setPort(remotePort);
-                logger.debug("Found SRV record \"" + remoteAddress + ":" + remotePort + "\"");
+        if (!(config instanceof GeyserPluginConfig)) {
```

Comment:
I would be very surprised if someone was using an SRV record to connect locally to the same server.

Reply:
sure, but if someone disabled direct connections and configures remote to their external IP (which they shouldn't do, but I've seen that a couple times..)

Then again, that's pretty niche.. but it would be difficult to discover as IPs are censored in geyser dumps. Maybe a warning if direct connection is disabled on plugin versions, with the remote address being a domain, could be justified?
dunno, it's definitely a cursed scenario

Reply:
Direct connection is left in as a courtesy at this point - once The Merge:tm: happens, it's planned to be the default, so we should probably stamp out these instances anyway.

Reply:
I can leave a log message in cases where it's disabled, though.